base_layer/core/src/covenants/covenant.rs

```diff
@@ -46,6 +46,8 @@ use crate::{
 const MAX_COVENANT_BYTES: usize = 4096;
 
 #[derive(Debug, Clone, PartialEq, Eq, Default)]
+/// A covenant allows a UTXO to specify some restrictions on how it is spent in a future transaction.
+/// See https://rfc.tari.com/RFC-0250_Covenants.html for details.
```

Comment:
Missing docs for public methods in `impl Covenant`, would be nice to add those as well